app/controllers/datasets_controller.rb

```diff
@@ -13,54 +23,75 @@ def create
     if @dataset.save
       redirect_to new_licence_dataset_path(@dataset)
     else
-      render 'new'
+      render 'dataset'
+    end
+  end
+
+  def update
+    @dataset = current_dataset
+    @dataset.update_attributes(params.require(:dataset).permit(:title, :summary, :description))
+
+    if @dataset.save
+      redirect_to dataset_path(@dataset)
+    else
+      render 'dataset'
     end
   end
 
   def licence
     @dataset = current_dataset
 
-    if request.post?
+    unless request.get?
       licence = get_licence(params.require(:dataset).permit(:licence, :licence_other))
       @dataset.licence = licence
 
-      redirect_to new_location_dataset_path(@dataset) if @dataset.save
+      if @dataset.save
+        redirect_to new_location_dataset_path(@dataset) if request.post?
+        redirect_to dataset_path(@dataset) if request.put?
+      end
     end
   end
 
   def location
     @dataset = current_dataset
 
-    if request.post?
+    unless request.get?
       location_params = params.require(:dataset).permit(:location1, :location2, :location3)
       @dataset.update_attributes(location_params)
 
-      redirect_to new_frequency_dataset_path(@dataset) if @dataset.save
+      if @dataset.save
+        redirect_to new_frequency_dataset_path(@dataset) if request.post?
+        redirect_to dataset_path(@dataset) if request.put?
+      end
     end
   end
 
   def frequency
     @dataset = current_dataset
 
-    if request.post?
+    unless request.get?
       @dataset.frequency = params.require(:dataset).permit(:frequency)[:frequency]
 
```

Comment:
Shouldn't be able to skip selecting a frequency, so might be easier to fix that way for now.